postem/postem-app/src/main/java/org/sakaiproject/postem/controller/UploadController.java

```diff
@@ -0,0 +1,177 @@
+/****************************************************************************** 
+* Copyright (c) 2020 Apereo Foundation
+
+* Licensed under the Educational Community License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+
+*          http://opensource.org/licenses/ecl2
+
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ ******************************************************************************/
+package org.sakaiproject.postem.controller;
+
+import java.text.MessageFormat;
+import java.util.Locale;
+import java.util.Objects;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.sakaiproject.api.app.postem.data.Gradebook;
+import org.sakaiproject.api.app.postem.data.StudentGrades;
+import org.sakaiproject.postem.constants.PostemToolConstants;
+import org.sakaiproject.postem.form.GradebookForm;
+import org.sakaiproject.postem.service.PostemSakaiService;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.tool.api.ToolSession;
+import org.sakaiproject.tool.cover.ToolManager;
+import org.sakaiproject.user.api.PreferencesService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Controller
+public class UploadController {
+
+	@Autowired
+    private PostemSakaiService postemSakaiService;
+    
+	@Autowired
+	private SessionManager sessionManager;
+	
+	@Autowired
+	private MessageSource messageSource;
+	
+    @Autowired
+    private PreferencesService preferencesService;
+	
+	private static final int TITLE_MAX_LENGTH = 255;
+	private static final int HEADING_MAX_LENGTH = 500;
+
+	@PostMapping(value = "/uploadFile")
+	public ResponseEntity<String> uploadFile(@RequestParam("file") MultipartFile file, HttpServletRequest request) {
+        log.debug("uploadFile");
+	    String userId = sessionManager.getCurrentSessionUserId();
+	    final Locale locale = StringUtils.isNotBlank(userId) ? preferencesService.getLocale(userId) : Locale.getDefault();
```

Comment:
Again, move this logic into BasePreferencesService

Reply:
Done.